code/game/turfs/open/ice.dm

```diff
@@ -55,11 +55,12 @@
 /turf/open/misc/ice/proc/dig_hole(mob/living/user)
 	if(!can_make_hole)
 		return FALSE
-	balloon_alert(user, "digging...")
-	playsound(src, 'sound/effects/shovel_dig.ogg', 50, TRUE)
-	if(!do_after(user, 5 SECONDS, src))
-		return FALSE
-	balloon_alert(user, "dug hole")
+	if(user)
```

Comment:
shouldn't this always be true?

Reply:
No, because we have a ice turf subtype that spawns with a dug hole. Thinking again though, I suppose I could wrap the remainder of the proc after these conditional statements into its own proc and call it instead. That way I shouldn't have to call `INVOKE_ASYNC` either.